medusa/providers/torrent/html/speedcd.py

```diff
@@ -183,20 +181,34 @@ def login(self):
             'password': self.password,
         }
 
-        for login_url in self.urls['login']:
-            response = self.session.post(login_url, data=login_params)
-            if not response or not response.text:
-                log.debug('Unable to connect to provider using login URL: {url}',
-                          {'url': login_url})
-                continue
+        if not (self.urls['login_post'] and self.login_url()):
+            log.debug('Unable get login URL')
+            return False
 
-            if re.search('Incorrect username or Password. Please try again.', response.text):
-                log.warning('Invalid username or password. Check your settings')
-                return False
-            return True
+        response = self.session.post(self.urls['login_post'], data=login_params)
+        if not response or not response.text:
+            log.debug('Unable to connect to provider using login URL: {url}',
+                      {'url': self.urls['login_post']})
+            return False
+        if re.search('Incorrect username or Password. Please try again.', response.text):
```

Comment:
You need a regex for this? Can't just do 'sring' in response.text ?

Reply:
No ideia. I just removed the for loop. need to check history